easybuild/easyconfigs/m/magma/magma-2.5.4-fosscuda-2019b.eb

```diff
@@ -12,16 +12,20 @@ toolchainopts = {'pic': True, 'openmp': True}
 
 source_urls = ['https://icl.cs.utk.edu/projectsfiles/magma/downloads/']
 sources = [SOURCE_TAR_GZ]
-checksums = ['7734fb417ae0c367b418dea15096aef2e278a423e527c615aab47f0683683b67']
+patches = ['magma-2.5.4_allow-all-sms.patch']
+checksums = [
+    '7734fb417ae0c367b418dea15096aef2e278a423e527c615aab47f0683683b67',  # magma-2.5.4.tar.gz
+    '7347bd27d008de4283db9212afe0eadaaa016cdc0100ede7646f45e5ccd583f2',  # magma-2.5.4_allow-all-sms.patch
+]
 
 builddependencies = [
     ('CMake', '3.15.3'),
 ]
 
-local_gpu_target = '-DGPU_TARGET="Kepler Maxwell Pascal Volta"'
+local_gpu_target = '%(cuda_sm_space_sep)s' or 'Kepler Maxwell Pascal Volta'
```

Comment:
This is not working without `cuda_compute_capabilities` being set. Instead of falling back to `Kepler Maxwell Pascal Volta`, the configuration has `-DGPU_TARGET="%(cuda_sm_space_sep)s"`.
I guess that in this case the template `cuda_sm_space_sep` has to be set to an empty string instead of being undefined.

Reply:
the issue might be that the templates are replaced later on, once the easyconfig is fully parsed. So this will never work as intended regardless of the contents of `%(cuda_sm_space_sep)s`.

Reply:
Hm, any idea how to generically solve this? This should likely go into the framework: Use build_option OR EC option for `cuda_compute_capabilities`. Currently this is implemented (twice!) in PyTorch and Tensorflow EasyBlocks and now we need it a third time.
ccing @boegel

Reply:
Oh, dismiss my last comment, I already implemented it like that. See the update to this PR, this should work now.
I only had to use the magma sm assigned values for the names